olioapps/typescript-starter-lab
# Added code and tests for event stream puzzler

**Ticket:**
N/a

**What this PR does**
This PR request adds the following
- Planning document with notes
- Converts class and methods code into single function, and two interfaces
- Jest for testing function

**Media**
![image](https://user-images.githubusercontent.com/7016167/173706221-5f8789a1-537c-4989-aacc-952385e5f7aa.png)


**How to test**
To test, clone the repository to your computer and navigate to the project with the terminal. Download all necessary packages with **_yarn install_** and when ready, run **_yarn run test_** to execute the application. You should see several tests and the results of these tests.

To use your own seed data, please enter the following at the bottom of index.ts, replacing [] with your event stream input array.

```
const seedData: ReadonlyArray<Event> = []
console.log(getWinningRegion(seedData))
```
Once you've added in your code, in the terminal run **_yarn run dev_** to see the output
